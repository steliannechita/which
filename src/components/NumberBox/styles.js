import styled from "styled-components";
import { media } from "../../utils";

export const NumberBoxWrapper = styled.button`
  ${media.desktop} {
    height: 60px;
    min-width: 190px;
  }
  ${media.tablet} {
    height: 60px;
    min-width: 285px;
  }
  ${media.mobile} {
    height: 60px;
    width: 85%;
    min-width: 240px;
  }
  border: 1px solid black;
  border-radius: 5px;
  margin: 5px;
  text-align: center;
  font-size: 22px;
  font-weight: 700;
  cursor: pointer;
  &.highlight {
    background-color: #fcf484;
  }
  &.with-border {
    border: 3px solid black;
  }
`;
