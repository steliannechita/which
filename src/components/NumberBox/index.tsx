import React from "react";
import { NumberBoxWrapper } from "./styles";

export interface NumberBoxProps {
  numberValue: number;
  isMultiple: boolean;
  isSelected: boolean;
  handleClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
}

const NumberBox = (props: NumberBoxProps) => {
  const { numberValue, isMultiple, isSelected, handleClick } = props;
  return (
    <NumberBoxWrapper
      value={numberValue}
      className={
        isSelected ? "with-border highlight" : isMultiple ? "highlight" : ""
      }
      onClick={handleClick}
    >
      {numberValue}
    </NumberBoxWrapper>
  );
};

export default NumberBox;
