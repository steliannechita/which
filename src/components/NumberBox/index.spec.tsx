import React from "react";
import { render } from "@testing-library/react";
import NumberBox from "./";

describe("NumberBox", () => {
  it("should render successfully with props", () => {
    const props = {
      numberValue: 1,
      isMultiple: false,
      isSelected: false,
      handleClick: (e: any) => e,
    };
    const { baseElement } = render(<NumberBox {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
