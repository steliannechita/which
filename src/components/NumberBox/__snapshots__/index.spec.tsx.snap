// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberBox should render successfully with props 1`] = `
<body>
  <div>
    <button
      class="sc-bdfBwQ bTeGaz"
      value="1"
    >
      1
    </button>
  </div>
</body>
`;
